### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by grace-mugoiri**

### CODE REVIEW

**Review by BalintMendli — CHANGES_REQUESTED** [review]
Hi @grace-mugoiri ,

Nice work on the project.

However, we need to sort out some issues.

First of all, this is supposed to be a collaborative project but I'm only seeing commits from one person. You might want to talk to Student Success about that.

### Linting

We have to set up linting on the project. Please install [stickler](https://github.com/microverseinc/linters-config/tree/master/javascript) on the repo and make sure it passes.

### Error after removing book

I'm getting this error when I try to change the read status of a book after removing a book:

error

Please fix this issue.

### UI

We also have to work on our UI. Divs overlapping and the content overflows the page:

ui

We should make it look nicer.

### Readme

Please add a descriptive readme for the project.

### Git flow

We are now using [Git flow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) on these projects. It would be difficult to change it on this repo but make sure you use it on your next projects.

When you're done making these changes please submit another Code Review Request.

**Comment by grace-mugoiri** [comment]
I am working alone on this project because there was no available partner for me and willow gave the go ahead.

**Review by BalintMendli — CHANGES_REQUESTED** [review]
Hi @grace-mugoiri ,

Great job, the UI looks much nicer now.

The error after deleting books still persist. We need to fix that. We want to modify books that do not exist anymore. Also we should make sure that we don't give the same id to different books.

Also stickler is not set up correctly yet. It is important to follow every step from https://github.com/microverseinc/linters-config/tree/master/javascript

Please make this changes and submit another Code Review Request.

If you have any questions do not hesitate to contact me on Slack.

**Review by JamesOkunlade — CHANGES_REQUESTED** [review]
### Hi @grace-mugoiri ✌️ Nice job making the corrections requested by the previous reviewer!!

Stickler is now working 😁but I'm sorry I can't give it a pass just yet till all linter checks are passing successfully. Please take some time more to check the linter check error details and refactor your code as recommended. 

Request for one last re-review when you're done 😁 

**Status: `changes required`**

**Happy coding!**

File: main.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+function showForm(){
+    if (formDiv.style.display === "none") { formDiv.style.display = "block"; }
+    if (addFormButton.style.display === "block") { addFormButton.style.display = "none"; }
+}
+addFormButton.addEventListener("click", showForm);
+
+function addBookToLibrary(book) {
+    myLibrary.push(book);
+}
+function resetForm(){
```
Comment (SunnyAsar) [inline main.js]: [OPTIONAL] you could avoid this whole method by using the `form` tag to wrap inputs and you can call `reset` on it to wipe it after submit

**Review by SunnyAsar — CHANGES_REQUESTED** [review]
# Hi @grace-mugoiri  👋 
### well done! the project looks good!
we can't move on because of the linting problems in the code as pointed out by @JamesOkunlade 
the errors are indentation, white spaces and using double quotes instead of single.
- I suggest installing a linter plugin for your IDE that can help catch these errors locally before you push, `ESlint` is a good candidate.
- Also, remove the comments in the code, the methods are well named.
**Please make a new request when its all smooth.** Cheers!

**Comment by grace-mugoiri** [comment]
I have really tried to fix these checks that are failing but to this point i may need more help on how to fix errors such as when to use double or single quotes and 4 spaces for tab instead of two.

**Comment by ghost** [comment]
Hi Grace, 
Your code is very great and understandable.
Just if you fix some list below, it will pass linter check and will look fine.
With my execution of your code, it has some way to pass linter check! :)
[It is the link for it](https://github.com/shjang7/test_linter/pulls)

[ For Javascript ]
There has an online tool for javascript linter: https://validatejavascript.com/
Before that, please remove the lines of 'no-undef' / 'quotes' / 'skipBlankLines' in eslint.config
And also you will need to add a top line in your main.js file for using (document.*) functions, that is
/* eslint-env browser */

Then, you can fix your code with eslint / the online linter above.
When you do fix as much as you can, then you can write some rule like 'no-alert' / 'func-names' in eslint.config (according to my execution of your code)
You could also reach to me with slack.

[ For HTML / CSS ]
And I will give you some suggestion about html view page.
1.
1st_improve_css
For the long size text, it is over the box like this. It doesn't look good.

2. 
2nd_improve_css
Between title and author name, it has no space between

3.
3rd_improve_css
To add book, it is overlay with other object, it doesn't look good.

4.
4th_improve_css
I can see the #container size is fixed with "width: 1400px". 
So, if someone sees with more smaller screens, it has left and right scroll.
Instead of that, it would be better to set with "min-width: 1400px" for now.
Optionally, it would be great to consider responsible for more smaller screens.

Please do consider javascript code, but also don't forget view is working fine at least.

**Comment by grace-mugoiri** [comment]
this time i got help from suhyeon who i got repaired with and passes all linters. 
Although my UI looks clustered in different pc if that is the case from the reviewer let me know.

**Review by AdnanAfsari — APPROVED** [review]
Hello @grace-mugoiri ✋ 

Great work 👏 , You have solved those linter errors that previous reviewer asked you to so its a 👍 from me, You can merge it now.
Status: Approved

Happy Coding

**Pull request merged**